MetadataExtractor/Formats/Png/PngMetadataReader.cs

```diff
@@ -167,15 +167,15 @@ private static IEnumerable<Directory> ProcessChunk(PngChunk chunk)
 
                     IccDirectory? iccDirectory = null;
                     Exception? ex = null;
-                    try
+
+                    if (TryDeflate(compressedProfile, 0, out var iccData, out var errorMessage))
                     {
-                        using var inflaterStream = new DeflateStream(new MemoryStream(compressedProfile), CompressionMode.Decompress);
-                        iccDirectory = new IccReader().Extract(new IndexedCapturingReader(inflaterStream));
```

Comment:
This now reads the entire compressed profile upfront, which may result in an additional allocation. This should be offset with the reduced indirection / direct access to the underlying buffer.